setup.md

```diff
@@ -0,0 +1,234 @@
+# Using Private Packagist in a Composer project
+
+You can use Private Packagist in a new or in an existing Composer project. Some of the steps in this guide will only apply if your project needs the respective Composer feature explained in that step.
+
+## Creating a new Composer project with Private Packagist
+
+When setting up a project it's important to add both the private repository URL for your Private Packagist organization and to disable packagist.org. Disabling the public open-source repository packagist.org is important because Private Packagist mirrors all the packages you get from there. If you do not disable it, Composer v1 will load all public packages twice which will slow down Composer and use more memory. Composer v2 automatically skips packagist.org packages if they are mirrored through Private Packagist.
+
+### composer init
+You can initialize a new Composer project using your Private Packagist repository by adding arguments to the `composer init` command.
+
+1. Configure authentication on your machine. The command to set an authentication token is listed on the organization overview.
+2. Assuming the organization URL name is `acme-company` run the command `composer init --repository="https://repo.packagist.com/acme-company/" --repository='{"packagist.org": false}'`
+
+### composer create-project
+If you would like to use the `create-project` command to initialize a project using a template package in your Private Packagist repository then these steps are required:
+
+1. Configure authentication on your machine. The command to set an authentication token is listed on the organization overview.
+2. Run `composer create-project --repository-url=https://repo.packagist/acme-company/`. This will create a new composer project for you.
+3. Once the command is finished you will still have to add your custom repository definition to the created composer.json file:
+
+```
+"repositories": [
+    {"type": "composer", "url": "https://repo.packagist.com/acme-company/},
+    {"packagist.org": false}
+]
+```
```

Comment:
I intentionally skipped that here because for now --add-repository does not add packagist.org: false, so it's really only suitable in a vendor context which should be documented separately I think?

Reply:
I did actually create a ticket to make this work better on Composer earlier today: https://github.com/composer/composer/issues/8853